server/server-proxy/src/main/java/software/amazon/smithy/java/server/ProxyService.java

```diff
@@ -81,31 +84,82 @@ private ProxyService(Builder builder) {
         }
         this.serviceErrorRegistry = registryBuilder.build();
         this.allOperations = new ArrayList<>();
-        for (var operation : TopDownIndex.of(model).getContainedOperations(service.getId())) {
-            String operationName = operation.getId().getName();
-            var function =
-                    new DynamicFunction(dynamicClient,
+        var containedOperations = TopDownIndex.of(model).getContainedOperations(service.getId());
+
+        for (var operation : containedOperations) {
+            if (operation.hasTrait(ProxyOperationTrait.class)) {
+                var proxyOperation = model.expectShape(
+                        operation.expectTrait(ProxyOperationTrait.class).getDelegateOperation(),
+                        OperationShape.class);
+                String operationName = proxyOperation.getId().getName();
+
+                if (!operations.containsKey(operationName)) {
+                    var serverOperation = createServerOperation(
+                            operation,
+                            proxyOperation,
                             operationName,
-                            schemaConverter,
+                            true,
                             model,
+                            service);
+                    addOperation(serverOperation, operationName);
+                }
+            }
+        }
+
+        for (var operation : containedOperations) {
+            if (!operation.hasTrait(ProxyOperationTrait.class)) {
```

Comment:
this can be an `else` in the first iteration over containedOperations

Reply:
Refactored code and added a comment to clarify why we are doing it this way.